lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -1355,10 +1357,56 @@ void OCLToSPIRVBase::visitCallScalToVec(CallInst *CI, StringRef MangledName,
     });
 }
 
+namespace {
+// Return true if any users of the CallInst use any of the constants
+// introduced by the SPV_EXT_image_raw10_raw12 extension.
+bool usesSpvExtImageRaw10Raw12Constants(const CallInst *CI) {
+  const std::array ExtConstants{
+      OCLImageChannelDataTypeOffset + ImageChannelDataTypeUnsignedIntRaw10EXT,
+      OCLImageChannelDataTypeOffset + ImageChannelDataTypeUnsignedIntRaw12EXT};
+
+  // The return values for `OpImageQueryFormat` added by the extension are
+  // integer constants that may appear anywhere in LLVM IR.  Only detect some
+  // common use patterns here.
+  for (auto *U : CI->users()) {
+    for (auto C : ExtConstants) {
+      ICmpInst::Predicate Pred;
+      if (match(U, m_c_ICmp(Pred, m_Value(), m_SpecificInt(C)))) {
+        return true;
+      }
+      if (auto *Switch = dyn_cast<SwitchInst>(U)) {
+        if (any_of(Switch->cases(), [C](const auto &Case) {
+              return Case.getCaseValue()->equalsInt(C);
+            })) {
+          return true;
+        }
+      }
+    }
+  }
+  return false;
+}
+} // anonymous namespace
+
 void OCLToSPIRVBase::visitCallGetImageChannel(CallInst *CI,
                                               StringRef DemangledName,
                                               unsigned int Offset) {
   assert(CI->getCalledFunction() && "Unexpected indirect call");
+
+  if (Offset == OCLImageChannelDataTypeOffset) {
```

Comment:
It seems like we don't check if the extension was enabled by the frontend driver. Should we do this check and report an error if it's not enabled?